chia/rpc/full_node_rpc_api.py

```diff
@@ -702,9 +701,10 @@ async def get_puzzle_and_solution(self, request: Dict) -> EndpointResult:
         if error is not None:
             raise ValueError(f"Error: {error}")
 
-        puzzle_ser: SerializedProgram = SerializedProgram.from_program(Program.to(puzzle))
-        solution_ser: SerializedProgram = SerializedProgram.from_program(Program.to(solution))
```

Comment:
avoiding these conversions look like an optimization, does it affect correctness?

Reply:
`Program.to()` does not support converting from `SerializedProgram`